src/Rings/mpoly.jl

```diff
@@ -661,15 +661,15 @@ function singular_assure(I::IdealGens, ordering::MonomialOrdering)
   else
       #= singular ideal exists, but the singular ring has the wrong ordering
        = attached, thus we have to create a new singular ring and map the ideal. =#
-      if !isdefined(I, :ord) || I.ord != ordering.o
+      if !isdefined(I, :ord) || I.ord != ordering
```

Comment:
You can intermediately fix it in this way, still we need to fix #1914 (hopefully via #2109).

Reply:
As far as I understand it, the condition `I.ord != ordering.o` would always evaluate to `true` because it doesn't compare the right things. So it builds a new ring and a new ideal every time.